### DISCUSSION TITLE
Proposal: make all bundles public

### DISCUSSION THREAD

**Discussion by sbarratt** [flashbots/pm#66] [general]
As it stands, flashbot bundles are only broadcasted to a curated list of miners, which are trusted to not disseminate the bundles or do anything other than include the bundle as the first n transactions in the prescribed block. The main reason this is done is because the bundles are just raw signed transactions, and so by design are not yet atomic. This makes the system inaccessible to smaller miners and individuals/organizations wanting to log flashbots data. It also puts too much trust to flashbots employees and miners that have access to the flashbot bundles before they are mined on-chain.

The ideal system: There are plans in the future to make the bundle fully atomic, i.e., its contents are not known until it is mined on-chain. When this is the case, anyone (miners, searchers, researchers, blockchain explorers, etc...) should be able to subscribe to MEV bundles, because nothing about the contents can be known until they are mined on chain.

The proposed system in-between the current and ideal system: Since flashbots essentially serves as a PGA that uses the proceeds from a transaction to pay the gas price, make all bundles public to everyone. (As pointed out in https://github.com/flashbots/pm/discussions/48, this does not change the mechanics of the auction; the equilibrium in a first price closed or open auction with identical payoffs is the payoff.) This only screws over the sandwichers that don't have the right precautions in place. (Note that they can still be screwed over in the current system either by miners or flashbots employees leaking their transactions, and they can still be screwed over in the ideal system if an uncle exposes their transaction.) This is not an issue for the arbitrage searchers, since their txns if included elsewhere will simply silently fail. I also think this is a good step towards the ideal system, since in the ideal system, all bundles are public anyways. Of course, if this is done, all searchers should be notified in advance that their raw transactions will be more public than in the current "trust-based" system.

**Comment by jgorfenger** [comment]
I would like to see reasons why this proposal would benefit Flashbots's stakeholders, namely miners and searchers. Even if the equilibrium doesn't change, it significantly lowers the bar for attacks on searchers who require atomic bundles. And then comes the low-latency game in which only those with the most sophisticated hardware (and most money) can compete to bid up everyone else. This would make Flashbots less democratic in my opinion.

One thing you have failed to consider is the privacy implications of having all bundles be public at this very moment. Most searchers do not want researchers or potentially malicious adversaries analyzing their every move; there is a certain level of trust in how the Flashbots team handles information that gets completely destroyed with your proposal. Flashbots has yet to release a full privacy policy, but making all bundles public would in essence nullify that. There are **_so many_** security issues stemming from this as well, ones that I'd rather not explain in-depth in case this actually happens. Having a trusted intermediary for this application is a luxury searchers can't afford to lose (yet). Don't forget that a major design goal of Flashbots _from the start_ was to include pre-trade privacy.

A lot of your analysis of how this proposal changes the game is theoretical. Try to think more about the practical consequences beyond just "screwing over sandwichers" and consider the actual stakeholders.

In short, I **oppose** this proposal for the foreseeable future. I think Flashbots should focus on migrating to a better system directly (mev-sgx) instead of doing something like this which is doing nothing to better the system.

**Comment by thegostep** [comment]
Related proposal: https://github.com/flashbots/pm/discussions/59
